nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/flow/synchronization/StandardVersionedComponentSynchronizer.java

```diff
@@ -509,6 +509,8 @@ private String determineRegistryId(final VersionedFlowCoordinates coordinates) {
             } else {
                 return explicitRegistryId;
             }
+        } else {
+            explicitRegistryId = "1";
```

Comment:
On closer review of this method, it seems like a better approach would be to adjust the subsequent logic that attempts to determine the `registryId` based on the applicability of the storage location. This could be interrelated to the other change to the `registryUrl`. In other words, the `storageLocation` containing additional path information could be causing the failure to find an applicable Flow Registry Client.

Reply:
So your most recent comment on my PR led me down an interesting path, haha. I see what you mean about looking into the subsequent call using the storageLocation. What's interesting about that is Stateless flows always ever use the InMemoryFlowRegistry.class , and then therefore that call:
try {
    locationApplicable = flowRegistryClientNode.isStorageLocationApplicable(location);
} catch (final Exception e) {
    LOG.error("Unable to determine if {} is an applicable Flow Registry Client for storage location {}", flowRegistryClientNode, location, e);
    continue;
}

if (locationApplicable) {
    LOG.debug("Found Flow Registry Client {} that is applicable for storage location {}", flowRegistryClientNode, location);
    return flowRegistryClientNode.getIdentifier();
}
So for stateless, the locationApplication is always false InMemoryFlowRegistry:
@Override
    public boolean isStorageLocationApplicable(final FlowRegistryClientConfigurationContext context, final String location) {
        return false;
    }
So, basically the only way to fix stateless is to either keep the "1" for registryId fix, or try to set this isStorageLocationApplication method for the in memory registry to true....although I'm sure that will probably break other things. Do you have any opinion on that?

Reply:
Okay, so a better way to fix this going with your guidance on that storageLocation check is to add a check for the in memory registry client (which is used 100% of the time for stateless). Code is ready for re-review.